### DISCUSSION TITLE
Help Wanted: error mounting "/home/user/docker/shlink/servers.json"

### DISCUSSION THREAD

**Discussion by Shane091490** [shlinkio/discussions#12] [Help wanted] [answered]
### Shlink version

latest

### PHP version

idk

### How do you serve Shlink

Other (explain in summary)

### Database engine

MariaDB

### Database version

10.8

### Summary

Hi 
I am having issue with mounting the servers.json file. Docker is creating the servers.json as a directory instead of a file. I am using absolute path for it but still getting error. Not sure how to resolve this...
- Served locally on LAN for testing purposes.
- Using docker compose

version: "3"

services:
  shlink:
    image: shlinkio/shlink:stable
    restart: unless-stopped
    container_name: shlink-backend
    environment:
      - TZ="America/Chicago"
      - DEFAULT_DOMAIN=192.168.1.177:8987 #no http/https. no trailing slash
      - IS_HTTPS_ENABLED=false
      - GEOLITE_LICENSE_KEY=REDACTED #we'll need to get this key from maxmind.com
      - DB_DRIVER=maria
      - DB_USER=shlink
      - DB_NAME=shlink
      - DB_PASSWORD=REDACTED #change this
      - DB_HOST=database
    depends_on:
      - database
    ports:
      - 8987:8080

  database:
    image: mariadb:10.8
    restart: unless-stopped
    container_name: shlink-database
    environment:
      - MARIADB_ROOT_PASSWORD=REDACTED #change this
      - MARIADB_DATABASE=shlink
      - MARIADB_USER=shlink
      - MARIADB_PASSWORD=REDACTED #change this
    volumes:
      - ./db:/var/lib/mysql

  shlink-web-client:
    image: shlinkio/shlink-web-client
    restart: always
    container_name: shlink-gui
    volumes:
      - /home/user/docker/shlink/servers.json:/usr/share/nginx/html/servers.json #this file will be generated automatically
    depends_on:
      - shlink
    ports:
      - 8081:80

Error from docker: 
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/home/user/docker/shlink/servers.json" to rootfs at "/usr/share/nginx/html/servers.json": mount /home/user/docker/shlink/servers.json:/usr/share/nginx/html/servers.json (via /proc/self/fd/6), flags: 0x5000: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

**Comment by acelaya** [comment, marked as answer]
That volume is designed for you to provide servers to shlink-web-client, not the other way around.

See https://github.com/shlinkio/shlink-web-client/issues/1050#issuecomment-1921372761
